op-program: test flake

```
Failed
=== RUN   TestFastCannonBlockHeaderOracle_WithFallback
=== PAUSE TestFastCannonBlockHeaderOracle_WithFallback
=== CONT  TestFastCannonBlockHeaderOracle_WithFallback
    testlog.go:135: writing test log to /var/opt/circleci/data/workdir/tmp/testlogs/TestFastCannonBlockHeaderOracleWithFallback-16745.log
--- FAIL: TestFastCannonBlockHeaderOracle_WithFallback (1.41s)
panic: runtime error: slice bounds out of range [4205:4096] [recovered]
	panic: runtime error: slice bounds out of range [4205:4096]

goroutine 4766 [running]:
testing.tRunner.func1.2({0x151bdc0, 0xc000a18330})
	/data/mise-data/installs/go/1.22.7/src/testing/testing.go:1631 +0x24a
testing.tRunner.func1()
	/data/mise-data/installs/go/1.22.7/src/testing/testing.go:1634 +0x377
panic({0x151bdc0?, 0xc000a18330?})
	/data/mise-data/installs/go/1.22.7/src/runtime/panic.go:770 +0x132
bufio.(*Writer).Write(0xc000546a00, {0xc0015d2d80?, 0x2dd6cc0?, 0x15e9302?})
	/data/mise-data/installs/go/1.22.7/src/bufio/bufio.go:681 +0x1b5
github.com/ethereum/go-ethereum/log.(*TerminalHandler).Handle(_, {_, _}, {{0xc1f1ba6bac39b92e, 0x593df2d8, 0x2dd6cc0}, {0x15e9302, 0x13}, 0xfffffffffffffffc, 0x8c81aa, ...})
	/home/circleci/go/pkg/mod/github.com/ethereum-optimism/op-geth@v1.101503.2-rc.2/log/handler.go:80 +0x118
github.com/ethereum/go-ethereum/log.(*logger).Write(0xc00054a090, 0xfffffffffffffffc, {0x15e9302, 0x13}, {0xc000794440, 0x4, 0x4})
	/home/circleci/go/pkg/mod/github.com/ethereum-optimism/op-geth@v1.101503.2-rc.2/log/logger.go:173 +0x2a6
github.com/ethereum/go-ethereum/log.Debug({0x15e9302?, 0x13?}, {0xc000794440?, 0x4?, 0x4?})
	/home/circleci/go/pkg/mod/github.com/ethereum-optimism/op-geth@v1.101503.2-rc.2/log/root.go:55 +0x10f
github.com/ethereum/go-ethereum/core/rawdb.(*freezerTable).checkIndex(0xc000f16410, 0x6)
	/home/circleci/go/pkg/mod/github.com/ethereum-optimism/op-geth@v1.101503.2-rc.2/core/rawdb/freezer_table.go:547 +0x5ea
github.com/ethereum/go-ethereum/core/rawdb.(*freezerTable).repairIndex(0xc000f16410)
	/home/circleci/go/pkg/mod/github.com/ethereum-optimism/op-geth@v1.101503.2-rc.2/core/rawdb/freezer_table.go:401 +0x52
github.com/ethereum/go-ethereum/core/rawdb.(*freezerTable).repair(0xc000f16410)
	/home/circleci/go/pkg/mod/github.com/ethereum-optimism/op-geth@v1.101503.2-rc.2/core/rawdb/freezer_table.go:233 +0x1f2
github.com/ethereum/go-ethereum/core/rawdb.newTable({0xc0008de620, 0x63}, {0x15d84be, 0x8}, 0xc0009da280, 0xc0009da2d0, 0xc0008960e8, 0x77359400, 0x0, 0x0)
	/home/circleci/go/pkg/mod/github.com/ethereum-optimism/op-geth@v1.101503.2-rc.2/core/rawdb/freezer_table.go:193 +0x668
github.com/ethereum/go-ethereum/core/rawdb.NewFreezer({0xc0008de620, 0x63}, {0x15e5ff8, 0x11}, 0x0, 0x77359400, 0xc0003795f0)
	/home/circleci/go/pkg/mod/github.com/ethereum-optimism/op-geth@v1.101503.2-rc.2/core/rawdb/freezer.go:125 +0x60d
github.com/ethereum/go-ethereum/core/rawdb.newChainFreezer({0xc0008de620?, 0x5d?}, {0x15e5ff8?, 0x10?}, 0x10?)
	/home/circleci/go/pkg/mod/github.com/ethereum-optimism/op-geth@v1.101503.2-rc.2/core/rawdb/chain_freezer.go:67 +0xa5
github.com/ethereum/go-ethereum/core/rawdb.NewDatabaseWithFreezer({0x7d90e668a858, 0xc0006a27a0}, {0xc00063c4e0, 0x5d}, {0x15e5ff8?, 0x11?}, 0x0)
	/home/circleci/go/pkg/mod/github.com/ethereum-optimism/op-geth@v1.101503.2-rc.2/core/rawdb/database.go:196 +0xa5
github.com/ethereum/go-ethereum/node.openDatabase({{0x0, 0x0}, {0xc00063c420, 0x55}, {0xc00063c4e0, 0x5d}, {0x15e5ff8, 0x11}, 0x0, 0x0, ...})
	/home/circleci/go/pkg/mod/github.com/ethereum-optimism/op-geth@v1.101503.2-rc.2/node/database.go:54 +0xbe
github.com/ethereum/go-ethereum/node.(*Node).OpenDatabaseWithFreezer(0xc000639b20, {0x15dab72, 0x9}, 0x0, 0x0, {0x0, 0x0}, {0x15e5ff8, 0x11}, 0x0)
	/home/circleci/go/pkg/mod/github.com/ethereum-optimism/op-geth@v1.101503.2-rc.2/node/node.go:753 +0x318
github.com/ethereum/go-ethereum/eth.New(0xc000639b20, 0xc0001c2008)
	/home/circleci/go/pkg/mod/github.com/ethereum-optimism/op-geth@v1.101503.2-rc.2/eth/backend.go:140 +0x4e5
github.com/ethereum-optimism/optimism/op-program/client/l2/test.NewMiner(0xc0006a6000, {0x20830a0, 0xc0009c9110}, 0x0)
	/var/opt/circleci/data/workdir/op-program/client/l2/test/miner.go:74 +0x613
github.com/ethereum-optimism/optimism/op-program/client/l2.TestFastCanonBlockHeaderOracle_GetHeaderByNumber(0xc0006a6000)
	/var/opt/circleci/data/workdir/op-program/client/l2/fast_canon_test.go:22 +0x91
testing.tRunner(0xc0006a6000, 0x1ed2fd0)
	/data/mise-data/installs/go/1.22.7/src/testing/testing.go:1689 +0xfb
created by testing.(*T).Run in goroutine 1
	/data/mise-data/installs/go/1.22.7/src/testing/testing.go:1742 +0x390
panic: runtime error: slice bounds out of range [4205:4096] [recovered]
	panic: runtime error: slice bounds out of range [4205:4096]

goroutine 4768 [running]:
testing.tRunner.func1.2({0x151bdc0, 0xc000484108})
	/data/mise-data/installs/go/1.22.7/src/testing/testing.go:1631 +0x24a
testing.tRunner.func1()
	/data/mise-data/installs/go/1.22.7/src/testing/testing.go:1634 +0x377
panic({0x151bdc0?, 0xc000484108?})
	/data/mise-data/installs/go/1.22.7/src/runtime/panic.go:770 +0x132
bufio.(*Writer).Write(0xc000546a00, {0xc0015d2d80?, 0x2dd6cc0?, 0x15e9302?})
	/data/mise-data/installs/go/1.22.7/src/bufio/bufio.go:681 +0x1b5
github.com/ethereum/go-ethereum/log.(*TerminalHandler).Handle(_, {_, _}, {{0xc1f1ba6bac39d382, 0x593e0d2c, 0x2dd6cc0}, {0x15e9302, 0x13}, 0xfffffffffffffffc, 0x8c81aa, ...})
	/home/circleci/go/pkg/mod/github.com/ethereum-optimism/op-geth@v1.101503.2-rc.2/log/handler.go:80 +0x118
github.com/ethereum/go-ethereum/log.(*logger).Write(0xc00054a090, 0xfffffffffffffffc, {0x15e9302, 0x13}, {0xc000848380, 0x4, 0x4})
	/home/circleci/go/pkg/mod/github.com/ethereum-optimism/op-geth@v1.101503.2-rc.2/log/logger.go:173 +0x2a6
github.com/ethereum/go-ethereum/log.Debug({0x15e9302?, 0x13?}, {0xc000848380?, 0x4?, 0x4?})
	/home/circleci/go/pkg/mod/github.com/ethereum-optimism/op-geth@v1.101503.2-rc.2/log/root.go:55 +0x10f
github.com/ethereum/go-ethereum/core/rawdb.(*freezerTable).checkIndex(0xc000e880d0, 0x6)
	/home/circleci/go/pkg/mod/github.com/ethereum-optimism/op-geth@v1.101503.2-rc.2/core/rawdb/freezer_table.go:547 +0x5ea
github.com/ethereum/go-ethereum/core/rawdb.(*freezerTable).repairIndex(0xc000e880d0)
	/home/circleci/go/pkg/mod/github.com/ethereum-optimism/op-geth@v1.101503.2-rc.2/core/rawdb/freezer_table.go:401 +0x52
github.com/ethereum/go-ethereum/core/rawdb.(*freezerTable).repair(0xc000e880d0)
	/home/circleci/go/pkg/mod/github.com/ethereum-optimism/op-geth@v1.101503.2-rc.2/core/rawdb/freezer_table.go:233 +0x1f2
github.com/ethereum/go-ethereum/core/rawdb.newTable({0xc00063c2a0, 0x5f}, {0x15d84be, 0x8}, 0xc0009da280, 0xc0009da2d0, 0xc0006a0f58, 0x77359400, 0x0, 0x0)
	/home/circleci/go/pkg/mod/github.com/ethereum-optimism/op-geth@v1.101503.2-rc.2/core/rawdb/freezer_table.go:193 +0x668
github.com/ethereum/go-ethereum/core/rawdb.NewFreezer({0xc00063c2a0, 0x5f}, {0x15e5ff8, 0x11}, 0x0, 0x77359400, 0xc0003795f0)
	/home/circleci/go/pkg/mod/github.com/ethereum-optimism/op-geth@v1.101503.2-rc.2/core/rawdb/freezer.go:125 +0x60d
github.com/ethereum/go-ethereum/core/rawdb.newChainFreezer({0xc00063c2a0?, 0x59?}, {0x15e5ff8?, 0x10?}, 0x10?)
	/home/circleci/go/pkg/mod/github.com/ethereum-optimism/op-geth@v1.101503.2-rc.2/core/rawdb/chain_freezer.go:67 +0xa5
github.com/ethereum/go-ethereum/core/rawdb.NewDatabaseWithFreezer({0x7d90e668a858, 0xc000784300}, {0xc0009b43c0, 0x59}, {0x15e5ff8?, 0x11?}, 0x0)
	/home/circleci/go/pkg/mod/github.com/ethereum-optimism/op-geth@v1.101503.2-rc.2/core/rawdb/database.go:196 +0xa5
github.com/ethereum/go-ethereum/node.openDatabase({{0x0, 0x0}, {0xc0009b4300, 0x51}, {0xc0009b43c0, 0x59}, {0x15e5ff8, 0x11}, 0x0, 0x0, ...})
	/home/circleci/go/pkg/mod/github.com/ethereum-optimism/op-geth@v1.101503.2-rc.2/node/database.go:54 +0xbe
github.com/ethereum/go-ethereum/node.(*Node).OpenDatabaseWithFreezer(0xc00078e1c0, {0x15dab72, 0x9}, 0x0, 0x0, {0x0, 0x0}, {0x15e5ff8, 0x11}, 0x0)
	/home/circleci/go/pkg/mod/github.com/ethereum-optimism/op-geth@v1.101503.2-rc.2/node/node.go:753 +0x318
github.com/ethereum/go-ethereum/eth.New(0xc00078e1c0, 0xc000004388)
	/home/circleci/go/pkg/mod/github.com/ethereum-optimism/op-geth@v1.101503.2-rc.2/eth/backend.go:140 +0x4e5
github.com/ethereum-optimism/optimism/op-program/client/l2/test.NewMiner(0xc0006a6340, {0x20830a0, 0xc0005a0360}, 0x4)
	/var/opt/circleci/data/workdir/op-program/client/l2/test/miner.go:74 +0x613
github.com/ethereum-optimism/optimism/op-program/client/l2.TestFastCannonBlockHeaderOracle_WithFallback(0xc0006a6340)
	/var/opt/circleci/data/workdir/op-program/client/l2/fast_canon_test.go:122 +0x94
testing.tRunner(0xc0006a6340, 0x1ed2fc8)
	/data/mise-data/installs/go/1.22.7/src/testing/testing.go:1689 +0xfb
created by testing.(*T).Run in goroutine 1
	/data/mise-data/installs/go/1.22.7/src/testing/testing.go:1742 +0x390
```

From this CI run: https://app.circleci.com/pipelines/github/ethereum-optimism/optimism/85680/workflows/3ff764a6-b938-4395-b47a-aac0b2a3b26d/jobs/3394946/tests

--- a/op-program/client/l2/test/miner.go
+++ b/op-program/client/l2/test/miner.go
@@ -63,8 +63,7 @@ func NewMiner(t *testing.T, logger log.Logger, isthmusTime uint64) (*Miner, *cor
 		NoPruning:   true,
 	}
 	nodeCfg := &node.Config{
-		Name:    "l2-geth",
-		DataDir: t.TempDir(),
+		Name: "l2-geth",
 	}
 	n, err := node.New(nodeCfg)
 	require.NoError(t, err)

